sentry_sdk/integrations/django/caching.py

```diff
@@ -1,80 +1,138 @@
     from sentry_sdk.integrations.django import DjangoIntegration
 
     original_method = getattr(cache, method_name)
 
     @ensure_integration_enabled(DjangoIntegration, original_method)
-    def _instrument_call(cache, method_name, original_method, args, kwargs):
-        # type: (CacheHandler, str, Callable[..., Any], Any, Any) -> Any
+    def _instrument_call(
+        cache, method_name, original_method, args, kwargs, address, port
+    ):
+        # type: (CacheHandler, str, Callable[..., Any], Any, Any, Optional[str], Optional[str]) -> Any
+        integration = sentry_sdk.get_client().get_integration(DjangoIntegration)
+        is_set_operation = method_name.startswith("set")
+        is_get_operation = not is_set_operation
+
+        op = OP.CACHE_SET_ITEM if is_set_operation else OP.CACHE_GET_ITEM
         description = _get_span_description(method_name, args, kwargs)
 
-        with sentry_sdk.start_span(
-            op=OP.CACHE_GET_ITEM, description=description
-        ) as span:
+        with sentry_sdk.start_span(op=op, description=description) as span:
             value = original_method(*args, **kwargs)
 
-            if value:
-                span.set_data(SPANDATA.CACHE_HIT, True)
-
-                size = len(str(value))
-                span.set_data(SPANDATA.CACHE_ITEM_SIZE, size)
-
-            else:
-                span.set_data(SPANDATA.CACHE_HIT, False)
+            with capture_internal_exceptions():
+                if address is not None:
+                    # Strip eventually contained username and password in url style location
+                    if "://" in address:
+                        parsed_url = urlparse(address)
+                        address = "{}://{}{}".format(
+                            parsed_url.scheme or "",
+                            parsed_url.netloc or "",
+                            parsed_url.path or "",
+                        )
+
+                    span.set_data(SPANDATA.NETWORK_PEER_ADDRESS, address)
+
+                if port is not None:
+                    span.set_data(SPANDATA.NETWORK_PEER_PORT, int(port))
+
+                key = _get_key(args, kwargs)
+                if key != "":
+                    span.set_data(SPANDATA.CACHE_KEY, key)
+
+                item_size = None
+                if is_get_operation:
+                    if value:
+                        if integration.cache_spans_add_item_size:
+                            item_size = len(str(value))
+                        span.set_data(SPANDATA.CACHE_HIT, True)
+                    else:
+                        span.set_data(SPANDATA.CACHE_HIT, False)
+                else:
+                    if integration.cache_spans_add_item_size:
+                        item_size = len(str(args[1]))
+
+                if item_size is not None:
+                    span.set_data(SPANDATA.CACHE_ITEM_SIZE, item_size)
 
             return value
 
     @functools.wraps(original_method)
```

Comment:
```suggestion
    @ensure_integration_enabled(DjangoIntegration, original_method)
```

To be applied alongside my comment on (original version) line 41; also, likely makes sense to do in separate PR.